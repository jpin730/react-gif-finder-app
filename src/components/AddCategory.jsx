import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const inputMinLength = 2;
  const [inputValue, setInputValue] = useState("");
  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };
  const onSubmit = (event) => {
    event.preventDefault();
    const trimmedInputValue = inputValue.trim();
    if (trimmedInputValue.length < inputMinLength) return;
    onNewCategory(trimmedInputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder={`Find Gifs (min. ${inputMinLength} chars)`}
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
