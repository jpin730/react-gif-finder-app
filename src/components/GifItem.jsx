import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
  return (
    <div className="card" title={title}>
      <img src={url} alt={title} />
      <p data-testid="p-title">{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
