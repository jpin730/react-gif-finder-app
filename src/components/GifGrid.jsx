import { useEffect, useState } from "react";

import { getGifs } from "../helpers/getGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then(setImages);
  }, []);

  return (
    <>
      <h3>{category}</h3>

      <hr />

      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
