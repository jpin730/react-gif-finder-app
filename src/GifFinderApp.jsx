import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifFinderApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (category) => {
    if (categories.includes(category)) return;
    setCategories([category, ...categories]);
  };

  return (
    <>
      <h1>Gif Finder App</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category}>
          {category}
        </GifGrid>
      ))}
    </>
  );
};
