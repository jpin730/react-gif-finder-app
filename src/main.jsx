import React from "react";
import ReactDOM from "react-dom/client";
import { GifFinderApp } from "./GifFinderApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <GifFinderApp />
  </React.StrictMode>
);
