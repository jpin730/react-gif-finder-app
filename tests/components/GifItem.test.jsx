import { render, screen } from "@testing-library/react";

import { GifItem } from "../../src/components";

describe("GifItem", () => {
  const title = "mocked-title";
  const url = "http://mocked-url/";

  test("should match snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("should set correct url and alt", () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show title as a paragraph", () => {
    render(<GifItem title={title} url={url} />);

    const { innerHTML } = screen.getByTestId("p-title");

    expect(innerHTML).toContain(title);
  });
});
