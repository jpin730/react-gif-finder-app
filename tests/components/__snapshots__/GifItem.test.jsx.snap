// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should match snapshot 1`] = `
<div>
  <div
    class="card"
    title="mocked-title"
  >
    <img
      alt="mocked-title"
      src="http://mocked-url/"
    />
    <p
      data-testid="p-title"
    >
      mocked-title
    </p>
  </div>
</div>
`;
